rmf_traffic_ros2/src/rmf_traffic_ros2/schedule/YamlLogger.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <rmf_traffic_ros2/schedule/ParticipantRegistry.hpp>
+#include <fstream>
+#include <mutex>
+
+namespace rmf_traffic_ros2 {
+
+class YamlLogger::Implementation 
+{
+public:
+  
+  Implementation(std::string file_path):
+  _file_path(file_path)
+  {
+    _counter = 0;
```

Comment:
```suggestion
    _counter = 0;
    if (!std::filesystem::exists(file_path))
    {
      std::filesystem::create_directories(
            std::filesystem::absolute(file_path).parent_path());
      return;
    }
```

If the file doesn't exist, let's create the directory here (to avoid an exception later) and skip trying to read it.